### ISSUE TITLE
[FEATURE] - Applock with device security

### ISSUE THREAD

**Issue by NL-TCH** [zaneschepke/wgtunnel#217]
Can we have an option to only unlock with device credentials (like device fingerprint, device passcode, devicepassword)?

this is way more secure and dynamic than the 4 digit pin code.
something similar is done in the nextcloud android client application.

![image](https://github.com/zaneschepke/wgtunnel/assets/58732773/9d4c5ed5-51fd-4fca-bb8d-23f1399169ff)

**Comment by NL-TCH** [comment]
thank you very much for this application. It is way better than the standard wireguard app by a long shot!

**Comment by zaneschepke** [comment]
So, we already have device fingerprint/pin if you try to export configs or if you try to edit/view the primary key of a tunnel. The purpose of the pin was to essentially serve as a parental control as a way to prevent people who already have device fingerprint/pin access from changing the settings and to help with devices that do not have any kind of auth (like AndroidTV). 

Are you saying you would like a setting to be able to require device fingerprint/pin to access the app ui?

**Comment by NL-TCH** [comment]
yes, that is exactly what i mean (this shouldn't have to high of a priority though!)

**Comment by Hill-98** [comment]
Since the app is now only PIN-protected due to #933, should we reconsider using other unlocking methods?

**Comment by diegombeltran** [comment]
Also: is it possible to use a pin/other method when enabling the tunnel via Android quick bar shortcuts?
